packages/salesforcedx-apex-replay-debugger/src/core/logContext.ts

```diff
@@ -99,6 +113,88 @@ export class LogContext {
     );
   }
 
+  public getHeapDumps(): ApexHeapDump[] {
+    return this.apexHeapDumps;
+  }
+
+  public scanLogForHeapDumpLines(): boolean {
+    const heapDumpRegex = RegExp(/\|HEAP_DUMP\|/);
+    for (const line of this.logLines) {
+      if (heapDumpRegex.test(line)) {
+        const splitLine = line.split('|');
+        const heapDump = new ApexHeapDump(
+          splitLine[3] /* heapDumpId */,
+          splitLine[4] /* className */,
+          splitLine[5] /* namespace */,
+          Number(splitLine[6]) /* line */
+        );
+        this.apexHeapDumps.push(heapDump);
+      }
+    }
+    return this.apexHeapDumps.length > 0;
+  }
+
+  public async fetchOverlayResultsForApexHeapDumps(
```

Comment:
The actual method to fetch the heap dumps. I'd considered a bulk call however that would be a potentially hideous amount of combined information. Developer Tools, inspecting the responses from single heap dump calls already has issues displaying larger heap dumps. With 5 large sized heap dumps it takes about 1.5-2.2 seconds (total) on startup to pull them all to the client.